src/vs/workbench/browser/quickaccess.ts

```diff
@@ -51,14 +53,22 @@ export function getQuickNavigateHandler(id: string, next?: boolean): ICommandHan
 		quickInputService.navigate(!!next, quickNavigate);
 	};
 }
-export class EditorViewState {
+export class PickerEditorState extends Disposable {
 	private _editorViewState: {
 		editor: EditorInput;
 		group: IEditorGroup;
 		state: ICodeEditorViewState | IDiffEditorViewState | undefined;
 	} | undefined = undefined;
+	private openedEditors = new Set<EditorInput>(); // editors that were opened between set and restore
 
-	constructor(private readonly editorService: IEditorService) { }
+	constructor(@IEditorService private readonly editorService: IEditorService) {
+		super();
+		this._register(this.editorService.onDidCloseEditor((e) => {
+			if (this._editorViewState) {
+				this.openedEditors.delete(e.editor);
+			}
+		}));
+	}
 
```

Comment:
I think this may actually not be needed if we go with `IEditorService.closeEditors`

Reply:
I guess this would just be to ensure that the set doesn't get too large. Do you think there's value to that, or do you think we can just ignore it and let `closeEditors` do nothing with the editorInputs that are closed?

Reply:
I do not think thats a problem, editor inputs are very lightweight objects typically.